src/prefect/workers/base.py

```diff
@@ -1562,6 +1591,267 @@ async def _mark_flow_run_as_cancelled(
                 f"Flow run '{flow_run.id}' was deleted before it could be marked as cancelled"
             )
 
+    def _is_cancellation_enabled(self) -> bool:
+        """Check if worker-side cancellation handling is enabled."""
+        return get_current_settings().worker.enable_cancellation
```

Comment:
This method seems extraneous since it's only used once and it's only one line.

I'm guessing it's here because it gets mocked during tests. Instead of using `monkeypatch`, you can you the `temporary_settings` context manager to override settings values for tests.